lib/usb/usb_standard.c

```diff
@@ -301,11 +313,17 @@ static int usb_standard_set_interface(usbd_device *usbd_dev,
 				      struct usb_setup_data *req,
 				      uint8_t **buf, uint16_t *len)
 {
-	const struct usb_config_descriptor *cfx = &usbd_dev->config[usbd_dev->current_config - 1];
+	const struct usb_config_descriptor *cfx;
 	const struct usb_interface *iface;
 
 	(void)buf;
 
+	if (! usbd_dev->current_config) {
+		return USBD_REQ_NOTSUPP;
+	}
```

Comment:
usb_20.pdf (p259)
9.4.10 Set Interface
Default state: Device behavior when this request is received while the device is in the Default state is not specified.
Address state: The device must respond with a Request Error.
Configured state: This is a valid request when the device is in the Configured state.

if the usb stack is tested for usual bugs, this will come up very nicely
also, the behaviour has been clearly mentioned.